src/facade/dragonfly_connection.cc

```diff
@@ -769,6 +776,20 @@ auto Connection::IoLoop(util::FiberSocketBase* peer, SinkReplyBuilder* orig_buil
   do {
     FetchBuilderStats(stats_, orig_builder);
 
+    if (migration_request_) {
+      ProactorBase* dest = migration_request_;
+
+      dispatch_fb_.JoinIfNeeded();
```

Comment:
Does migrate suspend? If so, what if you join, suspend inside migrate, and then get a async message from another thread that will call SendAsync, it'll see the fiber is not started and it'll start it...

Please note SendAsync is only called from its native thread, in the case here I assume that it just scheduled an await right when you migrate

Also thinking about this, what if we want to send a async message, so we look up what thread the connection lives on, but once we actually make it into the await, it already moved threads...

Reply:
So yes of course we should not do this with monitoring or pubsub, but I also remember that ACL updates its user info with async messages

Reply:
Ah, but they don't determine the threads beforehand, it just iterates over the listener afair